### PR TITLE
fix: archive testnet verification

### PR DESCRIPTION
**Pull request by BabyScope**
# Description

Closes: #XXXX

---

## Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] read the [contribution guide](https://github.com/lavanet/lava/blob/main/CONTRIBUTING.md)
* [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title, you can find examples of the prefixes below:
    
* [x] confirmed `!` in the type prefix if API or client breaking change
* [x] targeted the `main` branch
* [x] provided a link to the relevant issue or specification
* [x] reviewed "Files changed" and left comments if necessary
* [x] included the necessary unit and integration tests
* [x] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
* [x] confirmed all CI checks have passed

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] confirmed all author checklist items have been addressed
* [x] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

## Summary by CodeRabbit

- **New Features**
	- Enhanced `block` API call parsing with new `pruning` and `pruning-archive` directives, improving parsing capabilities and instructions.

### CODE REVIEW

File: cookbook/specs/near.json [hunk 1]
```diff
@@ -492,6 +492,19 @@
                             },
                             {
                                 "name": "pruning",
+                                "parse_directive": {
```
Comment (Yaroms) [inline cookbook/specs/near.json:495]: we have a really easy solution for that.
example from evmos: https://github.com/lavanet/lava/blob/0a14d3465e9e1440a5717ea43f801f7f6aed4506/cookbook/specs/evmos.json#L1560

by defining only the "function_tag": "GET_BLOCK_BY_NUM" you dont need to redefine the whole parse directive.

**Review by Yaroms — DISMISSED** [review]

File: cookbook/specs/near.json (hunk 1 above)
Comment (BabyScope) [reply cookbook/specs/near.json:484]: Hi @Yaroms, i i edted like this on https://github.com/lavanet/lava/pull/1498/commits/3dfe7aff5f9dca8c046388c3ededde83e06447a9

Comment (Yaroms) [reply cookbook/specs/near.json:484]: Hey, i spoke with @omerlavanet, we don't have the code supporting what i said, can you revert the changes back to your previous solution?

**Comment by BabyScope** [comment]
Hi @Yaroms and @omerlavanet , i changed the verification of the mainnet. First it uses the get block by num with latest distance of 64800 blocks after that verificatio for archive to query the block number. Tested out with nodes that has access to archive as non archive.

File: cookbook/specs/near.json [hunk 2]
```diff
@@ -492,6 +501,19 @@
                             },
                             {
                                 "name": "pruning",
```
Comment (omerlavanet) [inline cookbook/specs/near.json:503]: let's do this also pruning-archive, and then it inherits the 
"pruning" verification for a regular node

**Review by omerlavanet — CHANGES_REQUESTED** [review]

File: cookbook/specs/near.json (hunk 2 above)
Comment (BabyScope) [reply cookbook/specs/near.json]: https://github.com/lavanet/lava/pull/1498/commits/992a1b3a1ca3b7c3de412acfc59099e95eaa0231

cahnged naming and it is working tested on my nodes.

**Review by omerlavanet — APPROVED** [review]

**Pull request merged**